### PR TITLE
Design doc for patch merging list of scalars

### PR DESCRIPTION
**Pull request by mengqiy**
Design doc for how to make patch work for merging list of scalars based on https://github.com/kubernetes/kubernetes/pull/35647#issuecomment-262351199

Related issue: #35163, #32398

Last attempt fix in #35647 has been reverted, since we want to use the approach proposed in this PR.

cc: @kubernetes/kubectl

### CODE REVIEW

**Comment by k8s-oncall** [comment]
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/37652)

**Comment by ymqytw** [comment]
@deads2k Thanks for your comments. I have updated the doc.

>Is the slash important?

slash is the format for finalizers, but is confusing. I remove them to avoid confusing others.

This doc is for merging list which works like an unordered set.

**Comment by thockin** [comment]
@deads2k is this clearer to you?  I understood it (I think), but maybe we need the words "a special prefix for the key" or something?

**Comment by deads2k** [comment]
> @deads2k is this clearer to you? I understood it (I think), but maybe we need the words "a special prefix for the key" or something?

Yes, this helps.  I'm still concerned that there may be significant holes in patching capability, but I understand what this update is trying to tell me now.

File: docs/devel/api-conventions.md [hunk 1]
```diff
@@ -563,6 +566,43 @@ Strategic Merge Patch also supports special operations as listed below.
 
 ### List Operations
 
+#### List of Scalars
+
+We have two merge strategy for list of scalars: replace and merge. Default strategy is replace,
```
Comment (kargakis) [inline docs/devel/api-conventions.md]: s/strategy/strategies/

**Comment by ymqytw** [comment]
Created kubernetes/community#140.
I will close this PR once kubernetes/community#140 merged.

**Comment by ymqytw** [comment]
Closing because kubernetes/community#140 has been merged.

**Pull request closed without merging**